src/main/java/com/yelp/nrtsearch/server/luceneserver/search/TerminateAfterWrapper.java

```diff
@@ -137,9 +130,15 @@ public void setScorer(Scorable scorer) throws IOException {
       public void collect(int doc) throws IOException {
         if (collectedDocCount.incrementAndGet() > terminateAfter) {
           terminatedEarly = true;
-          throw new CollectionTerminatedException();
         }
-        leafCollector.collect(doc);
+
+        if (!terminatedEarly) {
+          // If we haven't terminated early, or if the collector is a TopHitsCollector, collect the doc
+          leafCollector.collect(doc);
+        } else if (leafCollector instanceof SearchCollectorManager.SearchCollector.SearchLeafCollector) {
+          // If we haven't terminated early, or if the collector is a TopHitsCollector, collect the doc
+          leafCollector.collect(doc);
```

Comment:
SearchLeafCollector is just a simple wrapper of the lucene collectors. In our case, this else if condition is pretty much else if (True).

So with this pr, we are scoring all candidates. The terminatedEarly is ignored.

Reply:
Our requirements for this feature would be when totalhitsthresshold > terminateAfter
The collector keeps counting the totalhit. But it does not do the real scoring any more. Otherwise, we may hit latency performance issues.

Reply:
Thanks for review, will address these points